Python/bytecodes.c

```diff
@@ -2615,10 +2615,13 @@ dummy_func(
             STAT_INC(FOR_ITER, hit);
             PyListObject *seq = it->it_seq;
             if (seq == NULL || it->it_index >= PyList_GET_SIZE(seq)) {
```

Comment:
I think this needs an `... || index < 0` condition for the free-threaded build. Alternatively, you can use the unsigned comparison trick from `valid_index`.

```
if ((size_t)it->it_index >= (size_t)PyList_GET_SIZE(seq)) {
    ...
}
```